### PR TITLE
Chandrashekhar workflow

### PR DESCRIPTION
**Pull request by Chandrashekhar-Bhavsar**
Please Review My code @Srishtirupa3

### CODE REVIEW

File: test_GetWorkFlowApi.py [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+        # Send a POST request to the API endpoint to add the new user
+        r = requests.post(url=API_ENDPOINT_post, data=json.dumps(data), headers=headers)
+        # Print the response status code
+        print("response error code we got",r.status_code)
+        # Get the response content as JSON
+        pastebin_url = r.json()
+        # Print the JSON response
+        print("response content we got",pastebin_url)
+        print(" ")
+        # Assert that the response status code is 200 (OK)
```
Comment (Srishtirupa3) [inline test_GetWorkFlowApi.py:51]: Please Update the comment as per the response  code 400.

File: workflow.py [hunk 2]
```diff
@@ -81,18 +90,20 @@ def getworkflowussue():
...
-        logging.debug(dt_string + " User has made a call for get workfflow module api")
+        logging.debug(dt_string+" User has made a call for get workfflow module api")
         logging.debug(dt_string+" Inside the get workfflow api ")       
         data = request.get_json()
         logging.debug(dt_string+" payload recived from frontend is ",data)
         issue = data["issueid"]
         wfn = data["wfn"]
+        if (type(issue) is not int) or (type(wfn) is not str):
+            return jsonify({"message": "data format is not right "}), 400
         cursor = mydb.cursor()
```
Comment (Srishtirupa3) [inline workflow.py:101]: please give some blank lines for code readability.

File: Filter.py [hunk 3]
```diff
@@ -161,3 +162,84 @@ def IssueFilterationQuarterly():
     except Exception as e:
         return jsonify({"error": "bad values"}), 400
     
+
+def DetailedIssueFilteration():
+    try:
```
Comment (Srishtirupa3) [inline Filter.py:167]: add doc string

File: Filter.py [hunk 4]
```diff
@@ -161,3 +162,84 @@ def IssueFilterationQuarterly():
...
+           Issue_Week[key] += value
+        print(Issue_Week)
+        query5 = "SELECT i.project_id,count(i.issue_id) FROM issue_member i INNER JOIN task t ON i.issue_id = t.issue_id WHERE t.task_sd between %s and %s group by i.project_id;"
+        values = (str(last_quarter_date.date()),str(current_date.date()))
+        cursor.execute(query5, values)
+        Q_task=cursor.fetchall()
+        query5 = "SELECT i.project_id,count(i.issue_id) FROM issue_member i INNER JOIN defect d ON i.issue_id = d.issue_id where d.defect_sd  between %s and %s group by i.project_id;;"
+        values = (str(last_quarter_date.date()),str(current_date.date()))
+        cursor.execute(query5, values)
+        Q_defect=cursor.fetchall()
```
Comment (Srishtirupa3) [inline Filter.py:229]: Please give some spaces for code readability

**Comment by pratiksatham** [comment]
done

**Pull request merged**